src/lib/util/slugify.ts

```diff
@@ -1,8 +1,10 @@
-// very basic slugify, currently only in use in tests
+/// strip out everything except characters, digits,
+// and whitespace, then convert to Kebab case.
+// E.g. "Title 2: This time it's personal"
+// becomes "title-2-this-time-its-personal"
 export default (s: string) =>
   s
     .toLowerCase()
-    .replaceAll(/[^a-z\s-_]/g, "")
-    .trim()
-    .replaceAll(/\s+/g, " ")
-    .replaceAll(" ", "-");
+    .replaceAll(/[^a-z0-9\s-&_]/g, "")
+    .split(" ")
+    .join("-");
```

Comment:
This change has some improvements (notably, number handling) but also has a couple issues:

- `&` should _not_ be used in slugs. It works in this case because the slug is only used as part of the hash, but if this function was used to generate a route it would break in surprising ways.
- This function removes the replacement of multiple consecutive whitespaces with a single space, which can lead to doubled hyphens in the slug.
- This function removes the `.trim` call, which can lead to hyphens at the beginning or end of the slug.
- This function splits on individual spaces, not sequences of whitespace, which can lead to empty strings in the slug leading to doubled hyphens or retaining whitespace that is not a space in the slug.

Something like this should work:

```ts
export default (s: string) =>
  s
    .toLowerCase()
    .replaceAll(/[^a-z0-9\s-_]/g, "")
    .trim()
    .replaceAll(/\s+/g, "-")
```

Reply:
I modified this to have an additional `replaceAll` preceding the current one with `.replaceAll(/&/g, "and")`. This will preserve legibility of links using ampersands ([here's a few in the wild under "Resources](https://www.ldaf.la.gov/land/fire/safety/statewide-burn-ban-issued-due-to-concerning-dry-conditions)) without messing up any routing